packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_filter_dialog.dart

```diff
@@ -59,101 +64,76 @@ class _ClassFilterDialogState extends State<ClassFilterDialog> {
-                        '  - ${ClassFilter.coreLibrariesAlias} for core classes without package prefix\n'
-                        '  - ${ClassFilter.dartAndFlutterLibrariesAlias} for "dart." and "package:" libraries published by Dart and Flutter orgs.'),
-                  ],
-                  SizedBox(height: itemSpacing),
-                  RadioButton<ClassFilterType>(
-                    label: 'Show all classes',
-                    itemValue: ClassFilterType.all,
-                    groupValue: _type,
-                    onChanged: onTypeChanged,
-                  ),
-                  SizedBox(height: itemSpacing),
-                  RadioButton<ClassFilterType>(
-                    label: 'Show all classes except:',
-                    itemValue: ClassFilterType.except,
-                    groupValue: _type,
-                    onChanged: onTypeChanged,
-                  ),
-                  Padding(
-                    padding: EdgeInsets.only(left: textFieldLeftPadding),
-                    child: TextField(
-                      keyboardType: TextInputType.multiline,
-                      maxLines: null,
-                      controller: _except,
-                    ),
-                  ),
-                  SizedBox(height: itemSpacing),
-                  RadioButton<ClassFilterType>(
-                    label: 'Show only:',
-                    itemValue: ClassFilterType.only,
-                    groupValue: _type,
-                    onChanged: onTypeChanged,
-                  ),
-                  Padding(
-                    padding: EdgeInsets.only(left: textFieldLeftPadding),
-                    child: TextField(
-                      keyboardType: TextInputType.multiline,
-                      maxLines: null,
-                      controller: _only,
-                    ),
-                  ),
-                ],
-              ),
+    return StateUpdateDialog(
+      title: 'Filter Classes and Packages',
+      helpText: _helpText,
+      onResetDefaults: () =>
+          setState(() => _loadStateFromFilter(ClassFilter.empty())),
+      onApply: () {
+        final newFilter = ClassFilter(
+          filterType: _type,
+          except: _except.text,
+          only: _only.text,
+        );
+        widget.onChanged(newFilter);
+      },
+      onCancel: null,
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          RadioButton<ClassFilterType>(
+            label: 'Show all classes',
+            itemValue: ClassFilterType.all,
+            groupValue: _type,
+            onChanged: onTypeChanged,
+          ),
+          const SizedBox(height: defaultSpacing),
+          RadioButton<ClassFilterType>(
+            label: 'Show all classes except:',
+            itemValue: ClassFilterType.except,
+            groupValue: _type,
+            onChanged: onTypeChanged,
+          ),
```

Comment:
Create a helper method that takes a label and a ClassFilterType to reduce duplicated code between all these RadioButtons